compact_str/src/tests.rs

```diff
@@ -1017,3 +1017,33 @@ fn test_insert() {
         "\u{ffff}\u{ffff}\u{ffff}\u{ffff}\u{ffff}\u{ffff}\u{ffff}\u{ffff}",
     );
 }
+
+#[test]
+fn test_shrink_to_fit() {
```

Comment:
could you add a test that creates a string of length 16MB + 1, and then shrink it to say 1MB? And then another test where we create a string of say 20MB and then shrink it to a size of 20MB - 1?

I want to make sure we cover the case two cases:
1. shrinking from a "capacity on heap" -> "capacity inlined"
2. shrinking from a "capacity on heap" -> "capacity on heap"